### ISSUE TITLE
Improved spacing for melismas in non-English languages

### ISSUE THREAD

**Issue by danielgarthur** [neanes/neanes#512]
**Is your feature request related to a problem? Please describe.**
In English scores, when a melisma is detected, the layout processor places together neumes that make up the melismatic phrase (see line 1 in the image below). However, in non-English languages, such as Greek, melismata are written as repeated vowels. The application does not officially recognize these as melismata and extra space is placed between the neumes (see line 2).

<img width="172" alt="image" src="https://github.com/neanes/neanes/assets/4132779/d77da9b6-68e4-4352-9b52-0bf584127442">

Neanes needs a way of detecting such melismata, either automatically or through user input.

**Describe the solution you'd like**
One option may be the following. Add an option to Page Setup to switch to an alternate melisma style. In this other mode, instead of the user repeatedly typing out the melismatic vowel, the user could type an underscore, as in English. Neanes would repeat the last vowel, instead of drawing a line.

A potential issue with this is that it may require hard-coded language support for each language to determine which letters are vowels in that language, so that the correct letters are repeated. This isn't a show-stopper, though, since it would not be much work to support the most common languages used in Byzantine Chant.

Another issue is that it would make it difficult to mix English and non-English on the same page unless an additional concept were introduced, such as "sections" or "paragraphs" that could be styled individually.

Another potential issue is that I believe I've seen in Greek books cases where a melismatic vowel was skipped for spacing considerations. So we may need a way for a user to hide vowels or maybe set a vowel frequency so that Neanes can automatically omit vowels if they would be too close together.

There is also the case of the gorthmikon (n and double n) and chi symbols, which could be part of the melisma.

**Describe alternatives you've considered**
Another way of supporting the melismata could be that when an underscore is detected, we try to guess the vowel or vowels and place them in the first melismatic neume group. The user could then have the option of changing our guess, in which case, each subsequent neume group in the melisma would use that lyric. And perhaps the user could override any lyric in the melisma to add, for example, a chi.

**Comment by danielgarthur** [comment]
Another, perhaps simpler, option could be to forgo the melisma detection, and instead allow users to highlight two or more neumes and "group" them together, in which case melismatic conventions would be followed. 

This could make sense as an alternative if the spacing issue is rare enough that users wouldn't have to be forming groups constantly throughout the score, but only in some occasional cases such as the example in the original post.

**Comment by basil** [comment]
Agreed that this is needed and would likely require hard-coded language support. Fortunately most contemporary Romanian publications seem to follow the same general conventions as the 19th-century Greek publications, though obviously the vowels and dipthongs are different in each language.

> Another issue is that it would make it difficult to mix English and non-English on the same page unless an additional concept were introduced, such as "sections" or "paragraphs" that could be styled individually.

While implementing #90 over the holidays (code which I still need to clean up and publish) I also had the need to split up a score into "paragraphs" that could be passed into the Knuth-Plass line breaking routine. I ended up devising a simple set of rules for inferring implicit paragraphs from the existing object model, but an explicit paragraph concept in the model and UI would certainly be helpful. Another place where I think it would make more sense to specify things per-paragraph rather than per-document is the choice between Chrysanthine and 1888 style accidentals. In my opinion a separate ticket should be created to track the "paragraph" concept, as it is an important object model and UI change with implications on several other features.

> Another potential issue is that I believe I've seen in Greek books cases where a melismatic vowel was skipped for spacing considerations.

While this can occur in a variety of contexts, by far the most common case is when a neume is followed by an yporroe with a gorgon, as in your example above (in both heirmologic and sticheraric pieces). The concept of a vowel frequency seems too closely coupled with a first-fit line breaking algorithm, but I think adding a checkbox to hide vowels makes sense. To reduce friction in the common case, hiding the vowel could be the default for an yporroe with a gorgon.

I'm less enthusiastic about exposing the `isMelisma` flag in the UI, since this seems like it would constrain our ability to evolve the implementation in the future.
